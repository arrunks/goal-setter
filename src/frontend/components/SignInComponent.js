import React,{useRef} from 'react';
import { connect } from 'react-redux';
import { doSignIn } from '../redux/actions';
import {Form, Button, Container, Row, Col} from 'react-bootstrap';

const SignInComponent = ({history,doSignIn}) => {
    const emailRef = useRef('');
    const passRef = useRef('');

    const onSubmit = (e) =>{
        e.preventDefault();
        doSignIn(emailRef.current.value, passRef.current.value).then(() => {
            history.push('/dashboard');
        });
    };
    return (
        <Container>
            <Row>
                <Col>
                    <Form onSubmit={onSubmit}>
                    <Form.Group controlId="formBasicEmail">
                        <Form.Label>Email address</Form.Label>
                        <Form.Control type="email" ref={emailRef} placeholder="Enter email" />
                    </Form.Group>

                    <Form.Group controlId="formBasicPassword">
                        <Form.Label>Password</Form.Label>
                        <Form.Control type="password" ref={passRef} placeholder="Password" />
                    </Form.Group>
                    <Button variant="primary" type="submit">
                        Sign In
                    </Button>
                    </Form>
                </Col>
            </Row>
        </Container>
    )
}


export default connect(null,{doSignIn})(SignInComponent);