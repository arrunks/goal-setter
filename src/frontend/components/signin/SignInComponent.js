import React,{useRef} from 'react';
import { connect } from 'react-redux';
import { doSignIn } from 'Actions';
import {Form, Button, Container, Row, Col} from 'react-bootstrap';
import { Motivation } from  'Components/motivation';
import styles from './style.css';

export const SignInComponent = ({history,doSignIn}) => {
    const emailRef = useRef('');
    const passRef = useRef('');

    const onSubmit = (e) =>{
        e.preventDefault();
        doSignIn(emailRef.current.value, passRef.current.value).then(() => {
            history.push('/dashboard');
        });
    };
    return (
        <Container fluid>
            <Row>
                <Col md={8} className={styles.signinContainer}>
                    <Motivation/>
                </Col>
                <Col className="pt-4">
                    <Form onSubmit={onSubmit}>
                    <Form.Group controlId="formBasicEmail">
                        <Form.Label>Email address</Form.Label>
                        <Form.Control type="email" ref={emailRef} placeholder="Enter email" />
                    </Form.Group>

                    <Form.Group controlId="formBasicPassword">
                        <Form.Label>Password</Form.Label>
                        <Form.Control type="password" ref={passRef} placeholder="Password" />
                    </Form.Group>
                    <Button variant="primary" type="submit">
                        Sign In
                    </Button>
                    </Form>
                </Col>
            </Row>
        </Container>
    )
}


export default connect(null,{doSignIn})(SignInComponent);