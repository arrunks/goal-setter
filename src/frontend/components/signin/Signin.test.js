import React from 'react';
import { mount  } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { SignInComponent }  from './SignInComponent';

const buttonText = 'Sign In';
let wrapped = mount (<SignInComponent />);

describe('Signin', () => {
    it('should render Component correctly', () => {   
      expect(wrapped).toMatchSnapshot();
    });

    it('renders the signin button', () => { 
      expect(wrapped.find('button').text()).toEqual(buttonText);
    });
  });