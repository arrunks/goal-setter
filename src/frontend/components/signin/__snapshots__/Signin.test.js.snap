// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin should render Component correctly 1`] = `
<SignInComponent>
  <Container
    className="pt-4"
    fluid={false}
  >
    <div
      className="pt-4 container"
    >
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            className="border-right"
            md={8}
          >
            <div
              className="border-right col-md-8"
            >
              <Motivation>
                <p
                  className="quoteText"
                >
                  “If you want to be happy, set a goal that commands your thoughts, liberates your energy and inspires your hopes.” —Andrew Carnegie
                </p>
              </Motivation>
            </div>
          </Col>
          <Col>
            <div
              className="col"
            >
              <Form
                inline={false}
                onSubmit={[Function]}
              >
                <form
                  className=""
                  onSubmit={[Function]}
                >
                  <FormGroup
                    controlId="formBasicEmail"
                  >
                    <div
                      className="form-group"
                    >
                      <FormLabel
                        column={false}
                        srOnly={false}
                      >
                        <label
                          className="form-label"
                          htmlFor="formBasicEmail"
                        >
                          Email address
                        </label>
                      </FormLabel>
                      <FormControl
                        placeholder="Enter email"
                        type="email"
                      >
                        <input
                          className="form-control"
                          id="formBasicEmail"
                          placeholder="Enter email"
                          type="email"
                        />
                      </FormControl>
                    </div>
                  </FormGroup>
                  <FormGroup
                    controlId="formBasicPassword"
                  >
                    <div
                      className="form-group"
                    >
                      <FormLabel
                        column={false}
                        srOnly={false}
                      >
                        <label
                          className="form-label"
                          htmlFor="formBasicPassword"
                        >
                          Password
                        </label>
                      </FormLabel>
                      <FormControl
                        placeholder="Password"
                        type="password"
                      >
                        <input
                          className="form-control"
                          id="formBasicPassword"
                          placeholder="Password"
                          type="password"
                        />
                      </FormControl>
                    </div>
                  </FormGroup>
                  <Button
                    active={false}
                    disabled={false}
                    type="submit"
                    variant="primary"
                  >
                    <button
                      className="btn btn-primary"
                      disabled={false}
                      type="submit"
                    >
                      Sign In
                    </button>
                  </Button>
                </form>
              </Form>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</SignInComponent>
`;
